File: BlackJackInCpp/BlackJackInCpp/Game.cpp
#include "stdafx.h"
#include "Game.h"


Game::Game()
{
}

void Game::position() {
	cout << "Player's points and Cards: " << player1.getPoints() << endl;
	player1.myCards();

	cout << "Dealer's points and Cards: " << dealer.getPoints() << endl;
	dealer.myCards();

	//cout << "Bank's points and Cards: " << bank.getPoints() << endl;
	//bank.myCards();
}

int Game::getRand() {
	return ((rand() % 8) + (rand() % 3)*(rand() % 7) + (rand() % 2)) % 13;
}

void Game::flop() {//a klasszikus szabalyok szerint az oszto mindket lapja lathato

	for (int i = 0; i < 4; i++) {
		int temp = getRand();
		try {
			bank.removeCard(temp);
		}
		catch (...) { i--; }	//ez csak tobb jatekos eseten tortenhetne meg
		if (i < 2)
			player1.getNewCard(temp);
		else
			dealer.getNewCard(temp);
	}
	position();
}

int Game::hit() {
	int temp = getRand();
	try {
		bank.removeCard(temp);
	}
	catch (...) {//igen kicsi a valoszinusege, hogy elfogyjon egy lap, aztan ujra ugyan azt dobja 2x
		temp = getRand();
		bank.removeCard(temp);
	}
	player1.getNewCard(temp);
	position();
	return player1.getPoints();
}

void Game::dealersRound() {//az oszto kore;

	while (dealer.getPoints() < 17) {
		int temp = getRand();
		try {
			bank.removeCard(temp);
		}
		catch (...) {
			temp = getRand();
			bank.removeCard(temp);
		}
		dealer.getNewCard(temp);
	}
	position();
}

void Game::calculateWinner() {
	if (dealer.getPoints() <= 21 && dealer.getPoints() >= player1.getPoints())
		cout << "The Bank won" << endl;
	else
		cout << "You won!" << endl;
}

Game::~Game()
{
}

File: BlackJackInPy/Game.py
from Deck import *
from random import randint


class Game:

    def __init__(self):
        self.bank = Deck("Bank")
        self.dealer = Deck("Dealer")
        self.player1 = Deck("Player")

    def position(self):
        print("Player's points and cards: " + str(self.player1.get_points()))
        self.player1.my_cards()
        print("Dealer's points and cards: " + str(self.dealer.get_points()))
        self.dealer.my_cards()

    def flop(self):
        for index in range(4):
            temp = self.get_rand()
            self.bank.remove_card(temp)
            if index < 2:
                self.player1.get_new_card(temp)
            else:
                self.dealer.get_new_card(temp)
        self.position()

    def hit(self):
        temp = self.get_rand()
        try:
            self.bank.remove_card(temp)
        except RuntimeError:    #igen kicsi a valoszinusege, hogy elfogyjon egy lap, es meg ugyan azt is dobja 2x
            temp = self.get_rand()
            self.bank.remove_card(temp)
        self.player1.get_new_card(temp)
        self.position()
        return self.player1.get_points()

    def calculate_winner(self):
        if self.dealer.get_points() <= 21 and self.dealer.get_points() >= self.player1.get_points():
            print("The Bank won!\n")
        else:
            print("You won!\n")

    def dealers_round(self):
        while self.dealer.get_points() < 17:
            temp = self.get_rand()
            try:
                self.bank.remove_card(temp)
            except RuntimeError:
                temp = self.get_rand()
                self.bank.remove_card(temp)
            self.dealer.get_new_card(temp)
        self.position()

    def get_rand(self):
        return randint(0, 12)

